src/System.Xml.XmlSerializer/src/System/Xml/Serialization/XmlSerializationWriterILGen.cs

```diff
@@ -2425,6 +2486,7 @@ internal void ILGenForCreateInstance(CodeGenerator ilg, Type type, Type cast, bo
             ilg.Call(Activator_CreateInstance);
             if (cast != null)
                 ilg.ConvertValue(Activator_CreateInstance.ReturnType, cast);
+            ilg.MarkLabel(labelReturn);
```

Comment:
I've found out that this path is also used for struct which doesn't have parameterless declared constructors. So the Activator.CreateInstance is needed.